dashboard/app/models/user_based_experiment.rb

```diff
@@ -0,0 +1,29 @@
+# == Schema Information
+#
+# Table name: experiments
+#
+#  id                     :integer          not null, primary key
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  name                   :string(255)      not null
+#  type                   :string(255)      not null
+#  start_time             :datetime
+#  end_time               :datetime
+#  section_id             :integer
+#  percentage             :integer
+#  earliest_section_start :datetime
+#  latest_section_start   :datetime
+#  script_id              :integer
+#
+
+class UserBasedExperiment < Experiment
+  def self.get_enabled(user: nil, section: nil, script: nil)
+    return Experiment.none unless user
+    Experiment.where(type: UserBasedExperiment.to_s).
+      where('percentage > (? + CONV(SUBSTRING(SHA1(name), 1, 10), 16, 10)) % 100', user.id)
```

Comment:
As you noted offline, though succinct, this seems awfully opaque. Could we define a mapping from strings to integers (it need not be random) and use the result of appending the mapping (on `experiments.name`) and `users.id` as the seed to a RNG?